src/Concordium/Client/Runner.hs

```diff
@@ -1750,19 +1751,40 @@ processModuleCmd action baseCfgDir verbose backend =
 
     ModuleList block -> do
       baseCfg <- getBaseConfig baseCfgDir verbose
-      (bestBlock, res) <- withClient backend $ withBestBlockHash block $ \bb -> (bb,) <$> getModuleList bb
-      v <- getFromJsonAndHandleError (\_ _ -> logFatal ["could not retrieve the list of modules",
-                                   "the provided block hash is invalid:", Text.unpack bestBlock]) $ grpcResponseVal <$> res
-      case v of
-        Nothing -> logFatal ["could not retrieve the list of modules",
-                               "the provided block does not exist:", Text.unpack bestBlock]
-        Just [] -> logInfo ["there are no modules in block " ++ Text.unpack bestBlock]
-        Just xs -> runPrinter $ printModuleList (bcModuleNameMap baseCfg) xs
+      -- Verify that the provided blockhash is valid.
+      case block of
+        Nothing -> return ()
+        Just b -> case readEither (Text.unpack b) of
+          Left _ -> logFatal ["Could not retrieve the list of modules",
+                              "the provided block hash is invalid, tried:",
+                              Text.unpack b]
```

Comment:
Won't this be on a separate line? Why?

Reply:
Yes, I do not know either. Probably it should just be a single line like everything else?

Reply:
Yes.